core/mock/mock.go

```diff
@@ -28,7 +28,13 @@ import (
 // NewMockNode constructs an IpfsNode for use in tests.
 func NewMockNode() (*core.IpfsNode, error) {
 	ctx := context.TODO()
-	nd := new(core.IpfsNode)
+	nd, err := core.NewIPFSNode(ctx, core.Offline(&repo.Mock{
```

Comment:
With this there is no need to expose full proc field to assign it to a new process.